COM-662 Impelement indicator 2b, 2c and 2d

lauravignoli opened the pull request.

lauravignoli committed:
COM-662 Impelement indicator 2b, 2c and 2d

tmvumbi2 commented on reportssql/treatment_report_functions.sql:
```diff
@@ -431,3 +434,37 @@ BEGIN
+            do.duration,
+            c.uuid -- uuid of the duration unit concept
+            ) BETWEEN timestampadd(MONTH, p_drugDurationMonthsMin, o.scheduled_date) 
+                AND timestampadd(MONTH, p_drugDurationMonthsMax, o.scheduled_date)
```
@lauravignoli the current implementation will not work well when the duration is a decimal.
If for example, we need to check that the duration is between 0 and 2 months; and there is a drug ordered on 10 Jan with a duration of 2 months and a half; the math will be:
- treatmentEndDate = 25 Mar (10 Jan + 2 months and 15 days)
- interval1 = 10 Jan (10 Jan + 0 month)
- interval2 = 10 Mar (10 Jan + 2 months)

25 Mar  BETWEEN 10 Jan AND 10 Mar => False

A possible fix is to create a function that calculates 'do.duration' to months (that discards the decimal, i.e. 2.8 months should return 2 months); and the condition could be something like:
``` getDurationInMonths(do.duration, c.uuid) BETWEEN p_drugDurationMonthsMin AND p_drugDurationMonthsMax```

lauravignoli replied:
@tmvumbi2 Why would we use the duration with decimals? If you want we can add a comment that the months must be integer values?

tmvumbi2 replied:
@lauravignoli sorry my comment wasn't clear enough, it was referring to the duration of the  treatment (i.e. not the interval provided as parameters)

tmvumbi2 commented on reportssql/treatment_report_functions.sql:
```diff
@@ -431,3 +434,37 @@ BEGIN
+    JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+    WHERE o.patient_id = p_patientId AND o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND calculateTreatmentEndDate(
```
@lauravignoli the duration needs to be checked on ARV satisfying the following conditions:
(1) if the current patient is an old patient: an ARV that was dispensed within the reporting period OR was dispensed before the reporting period but with a treatment that covers the entire reporting period
(2) if the current patient is not an old patient: an ARV that was dispensed within the reporting period

The function `isOldPatient(p_patient_id, p_startDate)` checks if a patient is old or not

lauravignoli replied:
Thanks, will fix it.

lauravignoli committed:
COM-626 Apply treatment duration on functions

Treatment duration was checked independently but it should be checked within the other indicator conditions.
